src/WorldClock.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Form, Button } from "react-bootstrap";
+
+class WorldClock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: <></>,
+    };
+    this.timezonesInput = React.createRef();
+    this.inputForm = (
+      <Form className="w-50" onSubmit={() => this.submit()}>
+        <Form.Group className="mb-3">
+          <Form.Label>Set-up your clocks</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Asia/Singapore America/Detroit"
+            ref={this.timezonesInput}
+          />
+        </Form.Group>
+        <Button variant="light" onClick={() => this.submit()}>
+          Show me the time!
+        </Button>
+      </Form>
+    );
+    this.state.display = this.inputForm;
+  }
+
+  submit() {
+    const timezonesInput = this.timezonesInput.current.value.trim().split(" ");
+    console.log(timezonesInput);
+    this.setState({
+      display: (
+        <>
+          <ClockTable timezonesInput={timezonesInput} />
+          <Button
+            variant="light"
+            onClick={() => this.setState({ display: this.inputForm })}
+          >
+            Take me back!
+          </Button>
+        </>
+      ),
+    });
+  }
+
+  render() {
+    return <>{this.state.display}</>;
+  }
+}
+
+export default WorldClock;
+
+class ClockTable extends React.Component {
+  //takes a list of timezones from props and uses the map function to generate multiple Row elements
+  clockItems = this.props.timezonesInput.map((timezone) => (
+    <Row className="justify-content-center align-items-center mb-3">
+      <Col xs lg="3">
+        {timezone}
+      </Col>
+      <Col xs lg="3" className="justify-content-center align-items-center">
+        <Clock timeZone={timezone} />
+      </Col>
+    </Row>
+  ));
+
+  render() {
+    return (
+      <Container>
+        <Row className="justify-content-md-center mb-3">
+          <Col xs lg="3">
+            Country
+          </Col>
+          <Col xs lg="3">
+            Time
+          </Col>
+        </Row>
+        {this.clockItems}
+      </Container>
+    );
+  }
+}
+
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { date: new Date() };
+  }
+  componentDidMount = () => {
+    this.timerId = setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  };
+  componentWillUnmount = () => {
+    clearInterval(this.timerId);
+  };
+  render() {
+    return (
+      <p className="m-0">
+        {this.state.date.toLocaleString("en-GB", {
+          timeZone: this.props.timeZone,
```

Comment:
I think this value does not need to come from a user input tbh. But rather should be a static display based on a time zone. If you want the user to define these, of course you can do it however - then you need to worry about where to store that input. 

Overall I feel there is a bit too much abstraction in this file. You could simplify this by taking away some layers, by defining what you want to achieve more clearly and rethinking how you want to utilize state.